packages/gatsby/src/schema/schema/index.js

```diff
@@ -13,13 +13,7 @@ const addResolvers = require(`./add-resolvers`)
 const addThirdPartySchemas = require(`./add-third-party-schemas`)
 
 const addTypeDefs = typeDefs => {
-  const types = schemaComposer.addTypeDefs(typeDefs)
-  types.forEach(type => {
-    const tc = schemaComposer.getTC(type.name)
```

Comment:
IIRC I kept this here (before `addInferredTypes`) so we can bail early there and not try to infer fields from `internal` etc.

Reply:
This forces schema composer too early, eg it fails if you refer to File type inside your typedefs when File type isn't inferred yet.